tests/tblgen_to_py/test.py

```diff
@@ -49,9 +49,7 @@ class Test_AnyOp(IRDLOperation):
 class Test_AttributesOp(IRDLOperation):
     name = "test.attributes"
 
-    int_attr = prop_def(
-        IntegerAttr[IntegerType].constr(type=EqAttrConstraint(IntegerType(16)))
-    )
+    int_attr = prop_def(IntegerAttr.constr(type=EqAttrConstraint(IntegerType(16))))
```

Comment:
If you wanted to simplify syntax further, you could allow this to take an attribute to be coerced into a Constraint

Reply:
I'd rather not go down that route, but it can be done in a different PR